chore(*) add go reports

gszr opened the pull request:
### Summary

Add anonymous reports for Go plugins:
- Go version in use
- Number of requests triggering a Go plugin
- Number of installed Go plugins

dndx commented on kong/db/dao/plugins.lua:
```diff
@@ -277,6 +282,10 @@ function Plugins:load_plugin_schemas(plugin_set)
       end
 
+      if handler._go then
+        go_plugins_cnt = go_plugins_cnt + 1
```
Can this be calculated above in the `load_plugin_handler` function.

gszr replied:
`load_plugin_handler` doesn't have a view of all plugins - it's called for each plugin - so `load_plugin_schemas` is the place where this sum can be done

dndx replied:
Cool, thanks for clarifying!

gszr commented:
@javierguerragiraldez Thanks for the review. Mind re-reviewing? I just pushed two changes: 1) extracted another helper (automatically building fixture go plugins and setting up the pluginserver path) and refactored the tests a bit - to avoid using `before_each` to start Kong and create entities. Affected commits were 7accb1a and 844906d.

javierguerragiraldez commented:
LGTM

gszr commented on spec/helpers.lua:
```diff
@@ -1697,13 +1697,41 @@ local function render_fixtures(conf, env, prefix, fixtures)
+      build_go_plugins(GO_PLUGIN_PATH)
+    end
+
+    if not env.go_pluginserver_exe and not os.getenv("KONG_GO_PLUGINSERVER_EXE") then
```
@javierguerragiraldez This is what I meant - only looking up in PATH if none of these have been set, as it'd otherwise potentially interfere with the use of KONG_GO_PLUGINSERVER_EXE in development workflows.

hishamhm replied:
Sorry, I commented over an outdated commit earlier.

I still think this is not really _necessary_, but since the environment variable is taking precedence in this version, I think it doesn't hurt either. So, :+1:

hishamhm commented on .requirements:
```diff
@@ -7,3 +7,4 @@ RESTY_LUAROCKS_VERSION=3.2.1
 RESTY_OPENSSL_VERSION=1.1.1d
 RESTY_PCRE_VERSION=8.43
 LIBYAML_VERSION=0.2.2
+KONG_GO_PLUGINSERVER_VERSION=master
```
I think we should tag a release number there and pin it here before merging.

gszr replied:
Done!
